extensions/amp-intaker-chat-widget/0.1/widget.js

```diff
@@ -0,0 +1,570 @@
+    showEl(frameContainer);
+    addClass(frameContainer, 'chatter-bot-frame-container-active');
+    addClass(launcherContainer, 'chatter-bot-frame-container-active');
+    if (!isDesktop) {
+      if (previewChatIsActive) {
+        addClass(frameContainer, 'preview');
+        addClass(launcherContainer, 'preview');
+      } else {
+        exitPreview();
+      }
+    }
+  }
+
+  /**
+   *
+   * @param {string} url
+   * @param {JsonObject} data
+   * @param {function(?): undefined} success
+   * @return {XMLHttpRequest}
+   */
+  function postAjax(url, data, success) {
+    const xhr = window.XMLHttpRequest
+      ? new XMLHttpRequest()
+      : new window.ActiveXObject('Microsoft.XMLHTTP');
+    xhr.open('POST', url);
+    xhr.onreadystatechange = function() {
+      if (xhr.readyState > 3 && xhr.status === 200) {
+        success(xhr.responseText);
+      }
+    };
+    xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.setRequestHeader('Accept', 'application/json');
+    xhr.send(JSON.stringify(data));
+    return xhr;
+  }
+
+  /**
+   *
+   * @param {boolean} [preview]
+   */
+  function lunch(preview) {
+    previewChatIsActive = false;
+    const dt = new Date();
+    dt.setMinutes(dt.getMinutes() + 15);
+    cookieValue.autoLunch = false;
+    CookiesAPI.set(cookieName, cookieValue, {expires: dt});
+
+    if (!chatIsActive) {
+      if (preview) {
+        previewChatIsActive = true;
+      } else {
+        chatIsActive = true;
+      }
+
+      if (!frameContainer) {
+        addToBody(Templates.frame);
+        frameContainer = document.getElementById(frameContainerId);
+        closeBtn = document.getElementById(closeBtnId);
+        restartBtn = document.getElementById(restartBtnId);
+        frame = document.getElementById(frameId);
+
+        hideEl(frameContainer);
+        closeBtn.addEventListener('click', closeFrame);
+        restartBtn.addEventListener('click', function() {
+          frame.src = url;
+        });
+        frame.src = url + (preview ? '&preview=true' : '');
+      }
+
+      openFrame();
+    } else {
+      closeFrame();
+    }
+  }
+
+  /**
+   *
+   */
+  function exitPreview() {
+    addClass(document.body, 'chatter-bot-body-noscroll');
+    removeClass(frameContainer, 'preview');
+    removeClass(launcherContainer, 'preview');
+    Intaker.SetStyle(frameContainer, 'height', '');
+    frame.contentWindow./*OK*/ postMessage('exitPreview', '*');
+  }
+
+  /**
+   *
+   * @param {string} url
+   * @return {string|null}
+   */
+  function getDirectLink(url) {
+    const parts = url.split('/');
+    while (true) {
+      let name = parts.pop();
+      if (name) {
+        name = name.trim();
+        if (name.length) {
+          return name;
+        }
+      } else {
+        return null;
+      }
+    }
+  }
+
+  /**
+   *
+   * @return {boolean}
+   */
+  function isMobile() {
+    let check = false;
```

Comment:
This should be reimplemented in terms of AMP's platform service

Reply:
there is no isMobile service just isAndroid and isIos, can I add isMobile function to the platform services?